### PR TITLE
[FrameworkBundle][HttpKernel] Add deprecation warning to show `HttpKernel::handle()` will catch throwables

### LINKED ISSUE #16205: [HttpKernel] Exceptions Responses are automatically sent
When an exception occurs in dev environment, the exception is catched and converted to a response which provides our nice symfony exceptions. 

However it is sent automatically by Symfony\Component\HttpKernel\HttpKernel::terminateWithException() which does:

```php
$response = $this->handleException($exception, $request, self::MASTER_REQUEST);
$response->sendHeaders();
$response->sendContent();

```

What is the reason for breaking the Request->Response workflow for exceptions ? You could  send the response and let the front controller call response->send() as usual. 

I am integrating a symfony app with a reactphp event loop. With such setup, I cannot send exceptions response to the client, they are printed to the console sadly displaying the whole markup. 

[edit] I tried to override the aforementioned method but since handleException is private, I cannot call it from my subclass.

My current option is to output to an error.html file and display it with my browser.

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | Fix #16205
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

I suggest that starting from Symfony 7.0 the `HttpKernel` will start to catch `\Throwable` and convert them to a `Response`.

This was first asked in #16205, I face a similar issue with Runtime component and Bref.. 

----------

The reason I push for this change is to embrace the request/response workflow of the Kernel without trusting the custom error handler. In an environment where you serve multiple requests with the same PHP process (read: RoadRunner, Swoole, Bref) you would write something like: 

```php
$kernel = new Kernel('prod', false);
while (true) {
  $request = /* create sf request from custom environment */
  try {
    $response = $kernel->handle(request);
    return ResponseConverter::convert($response);
  } catch (\Throwable $e) {
    exit(1);
  }
}
```
(pseudo code of course. Here is a [real example](https://github.com/php-runtime/bref/blob/0.3.1/src/BrefRunner.php#L30-L43))

The `exit(1)` means a hard crash. For Bref Runtime it would result in a 500 error from API-gateway. Since the `\Throwable` is caught, the Symfony error handler is not used. If we would not to catch the `\Throwable`, then the Symfony error handler would be used, but it would print a Response instead of returning it. (Printing a response will just add HTML on the CLI...)

-----------

Other PRs and issues related to this: 

- https://github.com/symfony/symfony/issues/45301
- https://github.com/symfony/symfony/pull/26514
- https://github.com/symfony/symfony/issues/22128
- https://github.com/symfony/symfony/pull/36885
- https://github.com/symfony/symfony/issues/25467

I'm happy to let the `HttpKernel` to catch the `\Throwable` exception right now, but I thought this very conservative PR would have a higher change to get merged. 

Also note that we do not specify any behaviour on our [HttpKernelInterface](https://github.com/symfony/symfony/blob/v6.0.7/src/Symfony/Component/HttpKernel/HttpKernelInterface.php#L43)

-------

To remove the deprecation message you need to add this to your config: 

```yaml
framework:
    catch_all_throwables: true
```

### CODE REVIEW

**Comment by stof** [comment]
A deprecation for which there is no way to opt-in for the new behavior is bad: it does not help migrating and will annoy people seeing it without being able to get rid of it.

**Comment by Nyholm** [comment]
Hm. You are very correct. Thank you. Do you have a suggestion on how to make an opt-in?

**Comment by stof** [comment]
Well, you would need some kind of configuration parameter that can turn on the catching of errors.

**Comment by Nyholm** [comment]
Thank you for the review. I've update the PR with a way to opt-in to the new behaviour. I've also added some tests.

**Comment by Nyholm** [comment]
Psalm and Fabbot are both false negatives. It seams like the 8.2 test fails for unrelated reasons. 

PHPUnit 8.1 low-deps fails because of this PR. Any suggestions how I could fix them?

**Comment by derrabus** [comment]
Thank you very much for bringing this topic up again. I like how you allow to opt-in to the new behavior.

> PHPUnit 8.1 low-deps fails because of this PR. Any suggestions how I could fix them?

The lazy fix would be to add conflict rules to the composer.json files of SecurityBundle and WebProfilerBundle, for FrameworkBundle < 6.1. The alterantive would be to make the test setup for the bundles more complex by setting/removing that option conditionally and I'm not sure that's worth it.

**Comment by Nyholm** [comment]
Thank you for the feedback. I've updated the PR.

**Comment by Nyholm** [comment]
The PR is rebased and updated. 

status: Waiting for review

**Comment by Nyholm** [comment]
The PR is updated for 6.2. Fabbot is a false negative and the PHP 8.2 test fail is unrelated. 

I would be happy to get another review on this. I believe it is ready to merge

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * This is useful since the HttpKernel catches all exceptions and turn them into
+ * an HTTP response.
+ *
+ * This class should only be used in tests.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class ExceptionListener
+{
+    private static array $exceptions = [];
```
Comment (derrabus) [inline src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php]:
```suggestion
    /**
     * @var list<\Throwable>
     */
    private static array $exceptions = [];
```

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    public function __invoke(ExceptionEvent $event)
+    {
+        self::$exceptions[] = $event->getThrowable();
+    }
+
+    public static function getExceptions(): array
```
Comment (derrabus) [inline src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php]:
```suggestion
    /**
     * @return list<\Throwable>
     */
    public static function getExceptions(): array
```

**Review by derrabus — APPROVED** [review]

**Comment by Nyholm** [comment]
Thank you for the review. I've applied the changes.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -133,6 +133,7 @@ public function getConfigTreeBuilder(): TreeBuilder
                 ->scalarNode('error_controller')
                     ->defaultValue('error_controller')
                 ->end()
+                ->booleanNode('catch_throwable_exception')->defaultFalse()->info('Tells HttpKernel to catch every kind of Throwable')->end()
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Maybe we could/should rename `catch_throwable_exception` to something like `catch_all_throwables` as not all throwables are exceptions.

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+
+/**
+ * This listener allows you to inspect all exceptions thrown by the application.
+ * This is useful since the HttpKernel catches all exceptions and turn them into
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php]:
```suggestion
 * This is useful since the HttpKernel catches all exceptions and turns them into
```

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        self::$exceptions[] = $event->getThrowable();
+    }
+
+    /**
+     * @return list<\Throwable>
+     */
+    public static function getExceptions(): array
+    {
+        $exceptions = self::$exceptions;
+        self::$exceptions = [];
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php]: For a getter method I would not expect the method to modify the object's state but to be able to call the method several times with the same result.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml [hunk 6]
```diff
@@ -5,3 +5,8 @@ framework:
     http_method_override: false
     uid:
         enabled: false
+
+services:
+    Symfony\Bundle\FrameworkBundle\Test\ExceptionListener:
+        tags:
+            - { name: kernel.event_listener, event: kernel.exception }
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml]: Would it make sense instead to let the `ExceptionListener` class implement the `EventSubscriberInterface`? That would make the registration in application easier.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/UidTest.php [hunk 7]
```diff
@@ -31,12 +32,16 @@ protected function setUp(): void
...
-        $this->expectException(\TypeError::class);
-        $this->expectExceptionMessage('Symfony\Bundle\FrameworkBundle\Tests\Functional\Bundle\TestBundle\Controller\UidController::anyFormat(): Argument #1 ($userId) must be of type Symfony\Component\Uid\UuidV1, string given');
-
         $client = $this->createClient(['test_case' => 'Uid', 'root_config' => 'config_disabled.yml']);
 
         $client->request('GET', '/1/uuid-v1/'.new UuidV1());
+        $this->assertSame(500, $client->getResponse()->getStatusCode());
+        $exceptions = ExceptionListener::getExceptions();
+        $this->assertCount(1, $exceptions);
+        $exception = $exceptions[array_key_first($exceptions)];
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/UidTest.php]:
```suggestion
        $exception = reset($exceptions);
```

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml (hunk 6 above)
Comment (Nyholm) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml]: It is just a class for testing. I think it should be opt-in.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php (hunk 3 above)
Comment (Nyholm) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Yeah. I was hesitant of this name too. I like your suggestion

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml (hunk 6 above)
Comment (xabbuh) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml]: It's still opt-in as it won't do anything if I don't register it as a service in my application, right? But assuming a standard Symfony application I would only have this configuration code instead:

```yaml
services:
    Symfony\Bundle\FrameworkBundle\Test\ExceptionListener: ~
```

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php (hunk 5 above)
Comment (Nyholm) [reply src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php]: I'll name this `resetExceptions()`

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml (hunk 6 above)
Comment (Nyholm) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml]: Thank you. 

I've updated the code

Comment (Nyholm) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml]: I had to make it: 

```yaml
services:
    Symfony\Bundle\FrameworkBundle\Test\ExceptionSubscriber:
        tags:
            - {name: kernel.event_subscriber}

```

Because EventSubscriberInterface was not automatically added the tag in this test env.

**Comment by Nyholm** [comment]
I've applied all changes and the CI is green

File: src/Symfony/Component/HttpKernel/HttpKernel.php [hunk 8]
```diff
@@ -82,6 +87,18 @@ public function handle(Request $request, int $type = HttpKernelInterface::MAIN_R
                 throw $e;
             }
 
+            return $this->handleThrowable($e, $request, $type);
```
Comment (HeahDude) [inline src/Symfony/Component/HttpKernel/HttpKernel.php]: Wouldn't it reduce the diff while keeping only one `catch` block to do something like this:
```php
try {
    // ...
} catch (\Throwable $e) {
    // only the first "if" needs to be removed in 7.0
    if (!$e instanceof \Exception && !$this->catchThrowable) {
        throw $e;
    }
    if ($e instanceof RequestExceptionInterface) {
        $e = new BadRequestHttpException($e->getMessage(), $e);
    }
    if (false === $catch) {
        $this->finishRequest($request, $type);

        throw $e;
    }

    return $this->handleThrowable($e, $request, $type);
}
```
?

File: src/Symfony/Component/HttpKernel/HttpKernel.php [hunk 9]
```diff
@@ -238,6 +255,12 @@ private function handleThrowable(\Throwable $e, Request $request, int $type): Re
         try {
             return $this->filterResponse($response, $request, $type);
         } catch (\Exception) {
+            return $response;
+        } catch (\Throwable $e) {
```
Comment (HeahDude) [inline src/Symfony/Component/HttpKernel/HttpKernel.php:249]: Same question here.

File: src/Symfony/Component/HttpKernel/HttpKernel.php (hunk 8 above)
Comment (Nyholm) [reply src/Symfony/Component/HttpKernel/HttpKernel.php]: Not by a hole not, no. I don’t think so. (I’ll try). 

I made two catch blocks to explicitly show this new code path. (And to be 100% sure I (and anybody after me) don’t make mistakes with the exception code path)

Comment (HeahDude) [reply src/Symfony/Component/HttpKernel/HttpKernel.php]: I added a comment in my example as I am talking about both the current diff and the one we'll get with 7.0.

Comment (Nyholm) [reply src/Symfony/Component/HttpKernel/HttpKernel.php]: Thank you.

**Comment by Nyholm** [comment]
I've applied the suggested changes. I've rebased the PR and the tests are green. 

Could I get a final round of reviews?

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionSubscriber.php [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+     */
+    public static function resetExceptions(): array
+    {
+        $exceptions = self::$exceptions;
+        self::$exceptions = [];
+
+        return $exceptions;
+    }
+
+    public static function getSubscribedEvents(): array
```
Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Test/ExceptionSubscriber.php:50]: Missing `{@inheritdoc}`?

File: src/Symfony/Component/HttpKernel/HttpKernel.php [hunk 11]
```diff
@@ -54,13 +54,18 @@ class HttpKernel implements HttpKernelInterface, TerminableInterface
...
-    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null, ArgumentResolverInterface $argumentResolver = null)
+    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null, ArgumentResolverInterface $argumentResolver = null, bool $catchThrowable = false)
     {
         $this->dispatcher = $dispatcher;
         $this->resolver = $resolver;
         $this->requestStack = $requestStack ?? new RequestStack();
         $this->argumentResolver = $argumentResolver ?? new ArgumentResolver();
+        $this->catchThrowable = $catchThrowable;
+        if (!$catchThrowable) {
+            trigger_deprecation('symfony/http-kernel', '6.2', 'Starting from 7.0, "%s::handle()" will catch \Throwable exceptions and convert them to HttpFoundation responses.', self::class);
```
Comment (HeahDude) [inline src/Symfony/Component/HttpKernel/HttpKernel.php]: This depreciation message is not actionable. It should tell to pass `true` as last argument to get rid of it.

Although this would still not be the right fix in the framework context, since the `catch_all_throwables` option needs to be set to `true`.
Maybe the configuration needs a `->validate()` node to trigger on `false` with the "good" message.

File: src/Symfony/Component/HttpKernel/HttpKernel.php [hunk 12]
```diff
@@ -54,13 +54,18 @@ class HttpKernel implements HttpKernelInterface, TerminableInterface
...
+    private bool $catchThrowable;
 
-    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null, ArgumentResolverInterface $argumentResolver = null)
+    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null, ArgumentResolverInterface $argumentResolver = null, bool $catchThrowable = false)
     {
         $this->dispatcher = $dispatcher;
         $this->resolver = $resolver;
         $this->requestStack = $requestStack ?? new RequestStack();
         $this->argumentResolver = $argumentResolver ?? new ArgumentResolver();
+        $this->catchThrowable = $catchThrowable;
```
Comment (HeahDude) [inline src/Symfony/Component/HttpKernel/HttpKernel.php:65]: Does it need to be deprecated as well, or it is allowed to be `false` in 7.0?
The current message sounds like it won't be supported anymore.

**Review by HeahDude — COMMENTED** [review]
~It would seem the `composer.json` file of the `WebProfilerBundle` needs an update as well (https://github.com/symfony/symfony/blob/6.2/src/Symfony/Bundle/WebProfilerBundle/composer.json#L21).~

EDIT: it seems I've missed https://github.com/symfony/symfony/pull/45997/files#diff-b7944e9265a9fd4f68f0fb505d9c8375791e607b9709dd3bd93519990683f4c7R62

**Comment by HeahDude** [comment]
It would be nice to have this default to `true` with the 6.2 recipe for new projects (and smooth upgrade with `composer recipes:update`).

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionSubscriber.php (hunk 10 above)
Comment (Nyholm) [reply src/Symfony/Bundle/FrameworkBundle/Test/ExceptionSubscriber.php:50]: Of the 80+ event subscribers, we only add this comment on a handful. I suggest to not include it. It brings little to no value.

File: src/Symfony/Component/HttpKernel/HttpKernel.php (hunk 12 above)
Comment (Nyholm) [reply src/Symfony/Component/HttpKernel/HttpKernel.php:65]: hm.. Im not sure we need to deprecate the property as it is part of the deprecation layer already.

File: src/Symfony/Component/HttpKernel/HttpKernel.php (hunk 11 above)
Comment (Nyholm) [reply src/Symfony/Component/HttpKernel/HttpKernel.php]: I've updated the exception message. 

> Maybe the configuration needs a ->validate() node to trigger on false with the "good" message.

Hm.. Yeah. But isn't that overkill?

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 13]
```diff
@@ -133,6 +133,7 @@ public function getConfigTreeBuilder(): TreeBuilder
                 ->scalarNode('error_controller')
                     ->defaultValue('error_controller')
                 ->end()
+                ->booleanNode('catch_all_throwables')->defaultFalse()->info('Tells HttpKernel to catch every kind of Throwable')->end()
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]:
```suggestion
                ->booleanNode('catch_all_throwables')->defaultFalse()->info('HttpKernel will catch all kinds of \Throwable')->end()
```
?

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionSubscriber.php [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * This event subscriber allows you to inspect all exceptions thrown by the application.
+ * This is useful since the HttpKernel catches all throwables and turns them into
+ * an HTTP response.
+ *
+ * This class should only be used in tests.
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/Test/ExceptionSubscriber.php:23]: Should we make it `@internal` or is it not needed in `\Test` sub namespace?

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml [hunk 15]
```diff
@@ -5,3 +5,8 @@ framework:
     http_method_override: false
     uid:
         enabled: false
+
+services:
+    Symfony\Bundle\FrameworkBundle\Test\ExceptionSubscriber:
+        tags:
+            - {name: kernel.event_subscriber}
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/Uid/config_disabled.yml]:
```suggestion
            - { name: kernel.event_subscriber }
```
As we use whitespaces on the import above 🤷‍♂️

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionSubscriber.php (hunk 14 above)
Comment (Nyholm) [reply src/Symfony/Bundle/FrameworkBundle/Test/ExceptionSubscriber.php:23]: I put it in Test namespace for two reasons: 
1) It is a nice small tool
2) I needed a way to figure out if FrameworkBundle was at version 6.2

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionSubscriber.php (hunk 10 above)
Comment (HeahDude) [reply src/Symfony/Bundle/FrameworkBundle/Test/ExceptionSubscriber.php:50]: Personally I like to use `{@inheritdoc}` since in reviews like this, it's the only way to differentiate methods that implement interfaces or override parents, from the others. (I agree I don't really need help for this scepcific method though, but I mean in general).

File: src/Symfony/Component/HttpKernel/HttpKernel.php (hunk 11 above)
Comment (HeahDude) [reply src/Symfony/Component/HttpKernel/HttpKernel.php]: I can see two paths. Once you get rid of the deprecation you should be ready for 7.0, but in this case it means passing `true` instead of `false`.

 - Removing the argument and the config node could be done in 8.0, once first deprecated in 7.1.

 - However removing the arg as it seems to be expected in 7.0 won't break, but will make passing the arg some dead code without notice. Also removing the config node in 7.0 would be a BC break if devs continue to define it to `true`.

File: src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php (hunk 5 above)
Comment (xabbuh) [reply src/Symfony/Bundle/FrameworkBundle/Test/ExceptionListener.php]: Now it's not really clear that I could use this method retrieve the list of all caught exceptions. What about adapting the schema used for methods in the SPL and use something like `shiftAll()`?

File: src/Symfony/Component/HttpKernel/HttpKernel.php (hunk 11 above)
Comment (Nyholm) [reply src/Symfony/Component/HttpKernel/HttpKernel.php]: Since the constructor argument and config is part of the deprecation layer they will both be removed in 7.0. 
Aren't we overthinking now?

File: src/Symfony/Component/HttpKernel/Tests/HttpKernelTest.php [hunk 16]
```diff
@@ -62,6 +77,80 @@ public function testHandleWhenControllerThrowsAnExceptionAndCatchIsTrueWithAHand
...
+    {
+        $dispatcher = new EventDispatcher();
+        $dispatcher->addListener(KernelEvents::EXCEPTION, function ($event) {
+            $event->setResponse(new Response($event->getThrowable()->getMessage()));
+        });
+
+        $kernel = $this->getHttpKernel($dispatcher, function () { throw new \TypeError('foo'); });
+        $this->expectException(\TypeError::class);
+        $response = $kernel->handle(new Request(), HttpKernelInterface::MAIN_REQUEST, false);
+        $this->assertEquals('foo', $response->getContent());
```
Comment (xabbuh) [inline src/Symfony/Component/HttpKernel/Tests/HttpKernelTest.php]: this statement cannot be reached

Comment (Nyholm) [reply src/Symfony/Component/HttpKernel/Tests/HttpKernelTest.php]: Thank you

File: src/Symfony/Component/HttpKernel/HttpKernel.php (hunk 11 above)
Comment (HeahDude) [reply src/Symfony/Component/HttpKernel/HttpKernel.php]: `choices_as_values` option was used for Symfony 2 deprecation layer, but deprecated in 3.1,
`logout_on_user_change` option was used in Symfony 3 deprecation layer and deprecated in 4.1 and so on.

The BC should be to not change your code when upgrading to 7.0, in this case changing `false` to `true` would make it work, but once done, another change would be required to remove config and arg, so they must still exist before we can safely remove them.

In other words, the deprecation layer itself cannot be deprecated in 6.x nor removed in 7.0, it needs to be deprecated in 7.1.

Comment (HeahDude) [reply src/Symfony/Component/HttpKernel/HttpKernel.php]: >Maybe the configuration needs a ->validate() node to trigger on false with the "good" message.

>Hm... Yeah. But isn't that overkill?

Well, the problem I see is that most devs using the framework do not even know there is an `HttpKernel` service nor how to change this arg.
However, mentioning the config key in this message from the component feels wrong.

**Comment by Nyholm** [comment]
I've rebased the PR and squashed the commits. 

The CI failures are not related to this PR. Could I ask you to review this PR again to decide if it is good enough to merge.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Nyholm.

**Comment by Nyholm** [comment]
Awesome. Thank you for merging. 

Big thank you to all reviewers too
